sinonjs/sinon
# Upgrade linting

This PR is a companion to https://github.com/sinonjs/eslint-config-sinon/pull/2

The branch makes a few changes:

* Upgrade ESLint to latest
* Fix all lint violations from new `eslint-config-sinon`
* Add Prettier
* Fix all style issues
* Add an npm task to run prettier
* Update the documentation
* Upgrade `eslint-config-sinon` to latest

Reviewers are encouraged to read commits individually, or suffer a `-6` roll for sanity.

#### How to verify - mandatory
1. In `@sinonjs/eslint-config-sinon`, check out the PR linked above
1. `npm link`
1. Switch to `@sinonjs/sinon`
1. Check out this branch
1. `npm install`
1. `npm link eslint-config-sinon`
1. Make some changes that violates the style rules
1. `npm run prettier -- --fix`
1. Observe that there are no more violations

#### Checklist for author
